recipes/packio/all/conanfile.py

```diff
@@ -0,0 +1,44 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class PackioConan(ConanFile):
+    name = "packio"
+    license = "MPL2"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/qchateau/packio"
+    description = "An asynchronous msgpack-RPC library built on top of Boost.Asio."
+    topics = ("rpc", "msgpack", "asio", "async", "cpp17")
+    settings = "compiler"
+    no_copy_source = True
+    requires = [
+        "msgpack/3.2.1",
+        "boost/1.72.0",
+    ]
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _supports_cpp17(self):
+        supported_compilers = [("apple-clang", 10), ("clang", 6), ("gcc", 7), ("Visual Studio", 15.7)]
+        compiler, version = self.settings.compiler, Version(self.settings.compiler.version)
+        return any(compiler == sc[0] and version >= sc[1] for sc in supported_compilers)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "packio-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        if not self._supports_cpp17():
```

Comment:
```suggestion
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, "17")
        elif not self._supports_cpp17():
```

Reply:
check cppstd when available, otherwise, check the compiler version.

Reply:
@uilianries Seems that compiler check is [already done by `check_min_cppstd`](https://github.com/conan-io/conan/blob/410be07a8666270ea79c0db181a48127251d1673/conans/client/tools/settings.py#L49), isn't it? I think the whole method should be removed and `check_min_cppstd` is already enough (the exception is raised too).

Reply:
`check_min_cppstd` seem to fallback on the _default_ `cppstd` of the compiler, but that does not mean that the compiler does not support the required `cppstd`.

Or am I wrong ?

Reply:
> `check_min_cppstd` seem to fallback on the _default_ `cppstd` of the compiler, but that does not mean that the compiler does not support the required `cppstd`.
> 
> Or am I wrong ?

It supports partial or mostly complete. https://en.cppreference.com/w/cpp/compiler_support

Reply:
> @uilianries Seems that compiler check and supported cpp standard deduction is [already done by `check_min_cppstd`](https://github.com/conan-io/conan/blob/410be07a8666270ea79c0db181a48127251d1673/conans/client/tools/settings.py#L49), isn't it? I think the whole method should be removed and `check_min_cppstd` is already enough (the exception is raised too).

No. check_min_cppstd checks the default cppstd supported, not the required.

Reply:
So is the current check the right one or do I need to change something ?

Reply:
It's okay, but I would recommend you follow this approach: https://github.com/conan-io/conan-center-index/blob/fb0a2b23e8978397a436fef3066e11117f7af66d/recipes/entt/3.x.x/conanfile.py#L30

Better messages and warning.

Reply:
> No. check_min_cppstd checks the default cppstd supported, not the required.

From the docs to `check_min_cppstd`:
> Check if current cppstd fits the minimal version required.

And `entt` library you mentioned uses exactly the same approach I'm talking about) - just use `check_min_cppstd` to require minimum C++ standard, and do not reinvent the wheel.